modules/internal/LastResortIO.chpl

```diff
@@ -0,0 +1,389 @@
+}
+*/
+pragma "last resort"
+proc stringStyleNullTerminated() {
+  compilerWarning("stringStyleNullTerminated will no longer be available by default,\nplease insert a use of the IO module to continue calling it.");
+  return stringStyleNullTerminated();
+}
+
+pragma "last resort"
+proc stringStyleExactLen(len:int(64)) {
+  compilerWarning("stringStyleExactLen will no longer be available by default,\nplease insert a use of the IO module to continue calling it.");
+  return stringStyleExactLen(len);
+}
+
+pragma "last resort"
+proc stringStyleWithVariableLength() {
+  compilerWarning("stringStyleWithVariableLength will no longer be available by default,\nplease insert a use of the IO module to continue calling it.");
+  return stringStyleWithVariableLength();
+}
+
+pragma "last resort"
+proc stringStyleWithLength(lengthBytes:int) throws {
+  compilerWarning("stringStyleWithLength will no longer be available by default,\nplease insert a use of the IO module to continue calling it.");
+  return stringStyleWithLength(lengthBytes);
+}
+
+pragma "last resort"
+proc IOHINT_NONE {
+  compilerWarning("IOHINT_NONE will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.IOHINT_NONE;
+}
+
+pragma "last resort"
+proc IOHINT_RANDOM {
+  compilerWarning("IOHINT_RANDOM will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.IOHINT_RANDOM;
+}
+
+pragma "last resort"
+proc IOHINT_SEQUENTIAL {
+  compilerWarning("IOHINT_SEQUENTIAL will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.IOHINT_SEQUENTIAL;
+}
+
+pragma "last resort"
+proc IOHINT_CACHED {
+  compilerWarning("IOHINT_CACHED will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.IOHINT_CACHED;
+}
+
+pragma "last resort"
+proc IOHINT_PARALLEL {
+  compilerWarning("IOHINT_PARALLEL will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.IOHINT_PARALLEL;
+}
+
+// TODO: iostyle record
+/*pragma "last resort"
+proc iostyle type {
+  compilerWarning("iostyle will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.iostyle;
+  }*/
+
+pragma "last resort"
+proc defaultIOStyle() {
+  compilerWarning("defaultIOStyle will no longer be available by default,\nplease insert a use of the IO module to continue calling it.");
+  return defaultIOStyle();
+}
+
+pragma "last resort"
+proc iohints type {
+  compilerWarning("iohints will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.iohints;
+}
+
+// TODO: file record methods
+pragma "last resort"
+proc file type {
+  compilerWarning("file will no longer be available by default, please\ninsert a use of the IO module to access it.");
+  return IO.file;
+}
+
+pragma "last resort"
+proc (IO.file).home ref: locale {
```

Comment:
Something about the ones like this one seems off/weird to me. E.g. if we resolved #14407 by saying that methods are available if a type is - wouldn't `myFile.home` already work?

This is part of the reason I think we should focus on making the wrapper-errors for types and free functions.